### PR TITLE
Eliminates AuthoredHintViewRequest and HintViewRequest lookups for non-CSF.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Note that, at present, proficiency is only tracked for CSF scripts. Thus, we need only lookup hint usage for these scripts.

We might also consider (re)asking product whether proficiency tracking needs to be done real-time or if it can be converted to a cron for performance reasons, noting that the initial spec required real-time tracking.

### CODE REVIEW

File: dashboard/app/models/user.rb [hunk 1]
```diff
@@ -903,8 +903,9 @@ def self.track_level_progress_sync(user_id:, level_id:, script_id:, new_result:,
 
       new_level_completed = true if !user_level.passing? &&
         Activity.passing?(new_result)
-      new_level_perfected = true if !user_level.perfect? &&
+      new_csf_level_perfected = true if !user_level.perfect? &&
         new_result == 100 &&
+        Script.get_from_cache(script_id).csf? &&
```
Comment (ashercodeorg) [inline dashboard/app/models/user.rb:908]: Should I guard against `Script.get_from_cache` returning `nil`?

**Comment by ashercodeorg** [comment]
CircleCI failure is legitimate (need to update tests), fixing in a followup commit.

File: dashboard/app/models/user.rb (hunk 1 above)
Comment (jeremydstone) [reply dashboard/app/models/user.rb:908]: I don't see any other place in our codebase where we do that, so I'd say no

**Comment by jeremydstone** [comment]
LGTM

**Pull request merged**